crates/tinymist-query/src/hover.rs

```diff
@@ -178,7 +178,21 @@ fn def_tooltip(
 
                         match &sym_docs {
                             Some(DefDocs::Variable(docs)) => {
-                                push_result_ty(def.name(), docs.return_ty.as_ref(), &mut type_doc);
+                                if let Some((_, long, _)) = docs.return_ty.as_ref() {
+                                    if long.as_str() == format!("@{}", def.name()) {
```

Comment:
well, you compared type by formatted result😄. The formatted type must not be used for equality checking, which is limitation to solve.